query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -189,6 +369,58 @@ public void handleRequest(
     }
   }
 
+  @Nonnull
+  private QueryRequest rewriteRequestWithViewFiltersApplied(QueryRequest request,
+      Map<String, ViewColumnFilter> columnFilterMap) {
+
+    Filter newFilter = removeViewColumnFilter(request.getFilter(), columnFilterMap);
+    QueryRequest.Builder builder = QueryRequest.newBuilder(request).clearFilter();
+    if (!Filter.getDefaultInstance().equals(newFilter)) {
+      builder.setFilter(newFilter);
+    }
+    return builder.build();
```

Comment:
Query should work just fine with a default empty filter (as that's what getFilter will always return anyway, never null)
```suggestion
    return QueryRequest.newBuilder(request).setFilter(newFilter).build();
```